debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java

```diff
@@ -502,22 +472,32 @@ public static void endMining(OracleConnection connection) {
      * @throws SQLException if anything unexpected happens
      */
     // todo: check RAC resiliency
-    public static void setRedoLogFilesForMining(OracleConnection connection, Long lastProcessedScn, Duration archiveLogRetention) throws SQLException {
+    public static void setRedoLogFilesForMining(OracleConnection connection, Scn lastProcessedScn, Duration archiveLogRetention) throws SQLException {
 
         removeLogFilesFromMining(connection);
 
-        Map<String, BigInteger> onlineLogFilesForMining = getOnlineLogFilesForOffsetScn(connection, lastProcessedScn);
-        Map<String, BigInteger> archivedLogFilesForMining = getArchivedLogFilesForOffsetScn(connection, lastProcessedScn, archiveLogRetention);
+        Map<String, Scn> onlineLogFilesForMining = getOnlineLogFilesForOffsetScn(connection, lastProcessedScn);
+        Map<String, Scn> archivedLogFilesForMining = getArchivedLogFilesForOffsetScn(connection, lastProcessedScn, archiveLogRetention);
 
         if (onlineLogFilesForMining.size() + archivedLogFilesForMining.size() == 0) {
             throw new IllegalStateException("None of log files contains offset SCN: " + lastProcessedScn + ", re-snapshot is required.");
         }
 
+        // Deduplicate log files with the same SCn ranges.
+        // todo: could this be eliminated by restricting the online log query to those there 'ARCHIVED="NO"'?
         List<String> logFilesNames = onlineLogFilesForMining.entrySet().stream().map(Map.Entry::getKey).collect(Collectors.toList());
-        // remove duplications
-        List<String> archivedLogFiles = archivedLogFilesForMining.entrySet().stream()
-                .filter(e -> !onlineLogFilesForMining.values().contains(e.getValue())).map(Map.Entry::getKey).collect(Collectors.toList());
-        logFilesNames.addAll(archivedLogFiles);
+        for (Map.Entry<String, Scn> archiveEntry : archivedLogFilesForMining.entrySet()) {
+            boolean found = false;
+            for (Map.Entry<String, Scn> redoEntry : onlineLogFilesForMining.entrySet()) {
+                if (redoEntry.getValue().equals(archiveEntry.getValue())) {
```

Comment:
Looks like we'd benefit from a bi-di map here? Perhaps also something for that follow-up above?